feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/domain/staking/start/setupStakingType/direct/DirectStakingTypeDetailsProvider.kt

```diff
@@ -0,0 +1,77 @@
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.common.types.StakingTypeDetailsProvider
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.common.types.StakingTypeDetailsProviderFactory
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.common.types.direct.StakingTypeDetailsInteractorFactory
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.setupAmount.SingleStakingProperties
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.setupAmount.SingleStakingPropertiesFactory
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.setupAmount.SingleStakingRecommendation
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.setupStakingType.model.EditableStakingType
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+class DirectStakingTypeDetailsProviderFactory(
+    private val stakingTypeDetailsInteractorFactory: StakingTypeDetailsInteractorFactory,
+    private val singleStakingPropertiesFactory: SingleStakingPropertiesFactory,
+    private val currentSelectionStoreProvider: StartMultiStakingSelectionStoreProvider
+) : StakingTypeDetailsProviderFactory {
+
+    override suspend fun create(
+        stakingOption: StakingOption,
+        coroutineScope: CoroutineScope,
+        availableStakingTypes: List<Chain.Asset.StakingType>
+    ): StakingTypeDetailsProvider {
+        val singleStakingProperties = singleStakingPropertiesFactory.createProperties(coroutineScope, stakingOption)
+        val validationSystem = ValidationSystem.editingStakingType(
+            singleStakingProperties,
+            availableStakingTypes
+        )
+        return DirectStakingTypeDetailsProvider(
+            validationSystem,
+            stakingTypeDetailsInteractorFactory.create(stakingOption, coroutineScope),
+            stakingOption.stakingType,
+            singleStakingProperties,
+            currentSelectionStoreProvider,
+            coroutineScope
+        )
+    }
+}
+
+class DirectStakingTypeDetailsProvider(
+    private val validationSystem: EditingStakingTypeValidationSystem,
+    stakingTypeDetailsInteractor: StakingTypeDetailsInteractor,
+    override val stakingType: Chain.Asset.StakingType,
+    private val singleStakingProperties: SingleStakingProperties,
+    private val currentSelectionStoreProvider: StartMultiStakingSelectionStoreProvider,
+    private val coroutineScope: CoroutineScope
+) : StakingTypeDetailsProvider {
+
+    override val recommendationProvider: SingleStakingRecommendation = singleStakingProperties.recommendation
+
+    override val stakingTypeDetails: Flow<EditableStakingType> = stakingTypeDetailsInteractor.observeData()
+        .map {
+            EditableStakingType(
+                isAvailable = validate() is ValidationStatus.Valid,
```

Comment:
We are already validation here so we can just store validationStatus directly instead of  `isAvaiable` flag. That way we dont need to re-run validation in ViewModel

Reply:
Need to discuss it.
It seems I should change ValidationExecutor to handle only Validation status without payload, progress etc.

Reply:
Agree, this wont be an easry change. Initially I didnt like creating validation payload in 3 different places but with StakingTypeDetailsProvider refactoring only 2 places will be left which is reasonable for given situation